vaadin-split-layout-flow/src/main/java/com/vaadin/flow/component/splitlayout/SplitLayout.java

```diff
@@ -349,9 +348,18 @@ public Registration addSplitterDragendListener(
         return super.addSplitterDragendListener(listener);
     }
 
-    private void setInnerComponentStyle(Component innerComponent,
-                                               String styleName, String value) {
-        Optional.ofNullable(innerComponent).ifPresent(component -> component
-                .getElement().getStyle().set(styleName, value));
+    private void setInnerComponentStyle(String styleName, String value,
+            boolean primary) {
+        Component innerComponent = primary ? primaryComponent
+                : secondaryComponent;
+        if (innerComponent != null) {
+            innerComponent.getElement().getStyle().set(styleName, value);
+        } else {
+            getElement().executeJs(
+                    "var children = this.children;"
+                            + "var element = children && children[$0];"
+                            + "if (element) { element.style[$1]=$2; }",
```

Comment:
nit: I feel slightly uncomfortable that the primary/secondary element selection is based on the index of the child element (0 or 1). The web component logic applies the `slot="primary"/"secondary"` attributes to the elements so it might be safer to use those instead (`var element = this.querySelector('[slot=$0]')` where $0 is either "primary" or "secondary"). If this causes issues with timing (web component hasn't applied the attributes in time), then it's not an option though.

Reply:
Initial implementation was with slot query and it didn't work. 
Did some extra tests now and slot is not set at the time the script is executed.
Is WC using the indexes to set the slots?

Reply:
[It seems to](https://github.com/vaadin/vaadin-split-layout/blob/master/src/vaadin-split-layout.html#L260). And the logic for that is invoked asynchronously by a FlattenedNodesObserver so it's a no-go. Invoking it synchronously first yourself with `this. _processChildren()` might be nastier than selecting the child by index.